### PR TITLE
[hermes] add /v2/updates/price/<timestamp> endpoint

### PR DESCRIPTION
**Pull request by cctdaniel**

### CODE REVIEW

File: hermes/src/api.rs [hunk 1]
```diff
@@ -27,7 +27,7 @@ use {
 mod doc_examples;
 mod metrics_middleware;
 mod rest;
-mod types;
+pub mod types;
```
Comment (cctdaniel) [inline hermes/src/api.rs:30]: reusing `PriceUpdate` in types for `state/benchmarks.rs`

File: hermes/src/state/benchmarks.rs [hunk 2]
```diff
@@ -56,22 +54,51 @@ impl TryFrom<BinaryBlob> for Vec<Vec<u8>> {
...
+            .parsed
+            .unwrap_or_default()
+            .into_iter()
+            .map(|parsed_price_update| PriceFeedUpdate {
+                price_feed:        PriceFeed::new(
+                    PriceIdentifier::new(hex::decode(&parsed_price_update.id).map(|v| {
+                        let mut arr = [0u8; 32];
+                        arr.copy_from_slice(&v);
+                        arr
+                    }).expect("Failed to decode hex string or output size does not match [u8; 32]")),
```
Comment (cctdaniel) [inline hermes/src/state/benchmarks.rs:71]: not sure this is the best way to do this

File: hermes/src/api/rest/v2/latest_price_updates.rs [hunk 3]
```diff
@@ -25,12 +25,13 @@ use {
...
     },
     pyth_sdk::PriceIdentifier,
+    serde::Deserialize,
     serde_qs::axum::QsQuery,
     utoipa::IntoParams,
 };
 
 
-#[derive(Debug, serde::Deserialize, IntoParams)]
+#[derive(Debug, Deserialize, IntoParams)]
```
Comment (cctdaniel) [inline hermes/src/api/rest/v2/latest_price_updates.rs:34]: drive-by: refactor

File: hermes/src/api/rest/v2/timestamp_price_updates.rs [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+    })?;
+
+    let price_update_data = price_feeds_with_update_data.update_data;
+    let encoded_data: Vec<String> = price_update_data
+        .into_iter()
+        .map(|data| match query_params.encoding {
+            EncodingType::Base64 => base64_standard_engine.encode(data),
+            EncodingType::Hex => hex::encode(data),
+        })
+        .collect();
```
Comment (ali-bahjati) [inline hermes/src/api/rest/v2/timestamp_price_updates.rs:126]: I think you can add a method to EncodingType like `encode_str` to encode a string and reuse it here and in latest_price_updates.

File: hermes/src/state/benchmarks.rs [hunk 5]
```diff
@@ -56,22 +55,50 @@ impl TryFrom<BinaryBlob> for Vec<Vec<u8>> {
     }
 }
 
-impl TryFrom<BenchmarkUpdates> for PriceFeedsWithUpdateData {
+
+impl TryFrom<PriceUpdate> for PriceFeedsWithUpdateData {
```
Comment (ali-bahjati) [inline hermes/src/state/benchmarks.rs:59]: I think this definition should be close to PriceUpdate.

nit: I do have a mixed feeling of having them all in the api module. we could entirely remove PriceFeedsWithUpdateData it it was not necessary for v1 api. Maybe let's move it after we deprecate v1 api.

**Review by ali-bahjati — APPROVED** [review]
Nice! Please address my comments before merging.

**Review by ali-bahjati — APPROVED** [review]
Nice! Please address my comments before merging.

File: hermes/src/state/benchmarks.rs (hunk 5 above)
Comment (cctdaniel) [reply hermes/src/state/benchmarks.rs:59]: hmm should we move it to `types.rs` instead then?
